feat: Bump example project to Angular 8

wtho opened the pull request:
I did most adjustments to be in sync with the default Angular CLI Boilerplate. I know many Angular projects will still differ from this, but I feel like it is the default starting point in the community. Open to discussion regarding this.

Changes in example project:
* Removed the whole e2e section in `angular.json`, as it looks quite different in current generated projects, but updating also the e2e folder seemed out of scope.
  **Question:** Should we remove the e2e source folder as well? Or keep everything also in `angular.json` in sync to the boilerplate of angular 8?
* I am not sure about the dependencies. Angular projects now lock the versions inside `package.json` using `~x.x.x`.
  **Question:** Should we keep all versions locked in the example project as well?
* Removed `core-js` from peerDependencies, it is not a dependency of Angular projects anymore, and made it a dependency of `jest-preset-angular`, as we need it for testing. See #266 for further discussion.
* Replaced `HttpModule` with `HttpClientModule` (`HttpModule` was dropped completely)
* `tsconfig.json` `baseUrl` is now the root folder, as it is in the Angular CLI boilerplate project
* `tsconfig.spec.json` needs `emitDecoratorMetadata: true` (or access to program/type checker through ts-jest), see #288 
* Adjustment to make `ng build --aot` not fail: ```[\`styles\`]: ...``` had to become `styles: ...` again and declared all components to `app.module`
* Updated `polyfills.ts` to current one
* Updated all ts config files
* Updated ts lint config file

Note that most tested files and tests remained untouched. Mostly the Angular Project files were changed.
Is this breaking due to moving `core-js` to dependencies?

Renaming the example project to e. g. `test-app` would be another task I would like to start with after this PR.

I would like to keep #288 open, as the cleaner approach might be a transformer (one day).

Solves #266.

wtho commented:
Just realized a changelog entry is still missing. Will add it soon!

ahnpnl commented:
- I agree e2e should be removed. It is irrelevant for the example app.
- I think the versions of dependencies for test app can just be kept in package.json as they are now.
- I think now we include core-js in dependencies is a breaking in the meaning of preset scope. In the meaning of usage scope, it's not really a breaking since it doesn't break anything on users' side unless users uninstalled core-js (I don't think there are any polyfill libraries better than it)

wtho commented:
There are polyfill libraries that are way more lightweight than `core-js` for the decorator metadata, but as it was a dependency of Angular for such a long time, we should still stick to it. Especially when the preset will be used with an older Angular version the users would have `core-js` installed already.

wtho commented:
Last Force Push
* Added Changelog information
* Removed the e2e folder and e2e configuration file
* Updated basic information of the example app `README.md` to inform users what the example app is

thymikee commented on CHANGELOG.md:
```diff
@@ -1,15 +1,19 @@
 * If the `astTransformers` are referenced in a custom `jest` config, `[ 'jest-preset-angular/build/InlineFilesTransformer', 'jest-preset-angular/build/StripStylesTransformer']` have to be set instead.
 * Serializers, transformers and `setupJest` have to be referenced from the `jest-preset-angular/build/`-folder in a custom config. Existing references have to be aligned.
 * If your `tsconfig.spec.json` is located in `src`, move it to your root folder and adjust the referenced files and paths inside, or align your jest configuration as discussed in the [README.md](https://github.com/thymikee/jest-preset-angular/blob/master/README.md#custom-tsconfig).
+* In an Angular 8 project or higher, `core-js` should no longer a dependency in the `package.json` of the Angular project, except if it is imported manually in the code.
```
```suggestion
* In an Angular 8 project or higher, `core-js` should no longer be a dependency declared in the `package.json`, unless it's imported manually in the code.
```

thymikee commented on example/tslint.json:
```diff
@@ -1,36 +1,49 @@
-  "rulesDirectory": [
-    "node_modules/codelyzer"
-  ],
+  "extends": "tslint:recommended",
```
lol, Angular still uses tslint? It's long been deprecated, we should use eslint instead, but if the default template comes with it, I'm ok

wtho commented:
Last Force Push:
* removed tslint dependency and configuration in the example app (was ignored and never run)
* rephrased changelog as recommended
